common/app/components/core/HtmlComponent.scala

```diff
@@ -0,0 +1,48 @@
+package components.core
+
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Document
+import org.jsoup.parser.Parser
+import play.twirl.api.Html
+import scala.collection.JavaConversions._
+
+trait HtmlComponent {
+
+  def html: Html
+  def componentCss: Seq[Css]
+
+  def render(): Html = {
+    val document: Document = Jsoup.parse(html.toString, "", Parser.xmlParser()) //Provide an XML Parser so Jsoup doesn't automatically add <html>, <head> and <body> tags
```

Comment:
As a general note, Jsoup is a Java library and therefore many/most of the methods provided can throw exceptions which will need to be handled here. For example `parse` throws an IOException, see: https://jsoup.org/apidocs/org/jsoup/Jsoup.html#parse-java.io.File-java.lang.String-

Simplest thing to do is to wrap in a Try.